GitUI/UserControls/RevisionGrid/Graph/LaneInfoProvider.cs

```diff
@@ -44,12 +44,30 @@ public string GetLaneInfo(int rowIndex, int lane)
                 laneInfoText.AppendLine(node.GitRevision.Guid);
 
                 var branch = new BranchFinder(node);
+                var mergedWith = branch.MergedWith;
                 if (branch.CommittedTo.IsNotNullOrWhitespace())
                 {
-                    laneInfoText.AppendFormat("\n{0}: {1}", Strings.Branch, branch.CommittedTo);
-                    if (branch.MergedWith.IsNotNullOrWhitespace())
+                    while (branch.IsMergedToMaster)
                     {
-                        laneInfoText.AppendFormat(MergedWithText.Text, branch.MergedWith);
+                        var leftmostChild = _nodeLocator.GetLeftmostNonartificialChild(branch.MergeNode);
+                        if (leftmostChild == null)
+                        {
+                            break; // from while
+                        }
+
+                        var childBranch = new BranchFinder(leftmostChild, branch.MergeNode);
+                        if (childBranch.CommittedTo.IsNullOrWhiteSpace())
+                        {
+                            break; // from while
+                        }
+
+                        branch = childBranch;
+                    }
+
+                    laneInfoText.AppendFormat("{0}{1}: {2}", Environment.NewLine, Strings.Branch, branch.CommittedTo);
```

Comment:
Is not git setting \n ?
The commit message and ge may not be in the same env, eg wsl